flagengine/engine_eval/evaluator.go

```diff
@@ -0,0 +1,321 @@
+		return s
+	}
+	// Handle *Value type
+	if v, ok := contextValue.(*Value); ok && v != nil {
+		if v.String != nil {
+			return *v.String
+		}
+		if v.Bool != nil {
+			return strconv.FormatBool(*v.Bool)
+		}
+		if v.Double != nil {
+			return strconv.FormatFloat(*v.Double, 'f', -1, 64)
+		}
+	}
+	return fmt.Sprint(contextValue)
+}
+
+func match(c Operator, traitValue, conditionValue string) bool {
+	// Handle special operators first
+	switch c {
+	case Modulo:
+		return matchModulo(traitValue, conditionValue)
+	case Regex:
+		return matchRegex(traitValue, conditionValue)
+	}
+
+	b1, e1 := strconv.ParseBool(traitValue)
+	b2, e2 := strconv.ParseBool(conditionValue)
+	if e1 == nil && e2 == nil {
+		return matchBool(c, b1, b2)
+	}
+
+	i1, e1 := strconv.ParseInt(traitValue, 10, 64)
+	i2, e2 := strconv.ParseInt(conditionValue, 10, 64)
+	if e1 == nil && e2 == nil {
+		return matchInt(c, i1, i2)
+	}
+
+	f1, e1 := strconv.ParseFloat(traitValue, 64)
+	f2, e2 := strconv.ParseFloat(conditionValue, 64)
+	if e1 == nil && e2 == nil {
+		return matchFloat(c, f1, f2)
+	}
+	if strings.HasSuffix(conditionValue, ":semver") {
+		conditionVersion, err := semver.Make(conditionValue[:len(conditionValue)-7])
+		if err != nil {
+			return false
+		}
+		return matchSemver(c, traitValue, conditionVersion)
+	}
+	return matchString(c, traitValue, conditionValue)
+}
+
+func matchSemver(c Operator, traitValue string, conditionVersion semver.Version) bool {
+	traitVersion, err := semver.Make(traitValue)
+	if err != nil {
+		return false
+	}
+	switch c {
+	case Equal:
+		return traitVersion.EQ(conditionVersion)
+	case GreaterThan:
+		return traitVersion.GT(conditionVersion)
+	case LessThan:
+		return traitVersion.LT(conditionVersion)
+	case LessThanInclusive:
+		return traitVersion.LTE(conditionVersion)
+	case GreaterThanInclusive:
+		return traitVersion.GE(conditionVersion)
+	case NotEqual:
+		return traitVersion.NE(conditionVersion)
+	}
+	return false
+}
+
+func matchBool(c Operator, v1, v2 bool) bool {
+	var i1, i2 int64
+	if v1 {
+		i1 = 1
+	}
+	if v2 {
+		i2 = 1
+	}
+	return matchInt(c, i1, i2)
+}
+
+func matchInt(c Operator, v1, v2 int64) bool {
+	switch c {
+	case Equal:
+		return v1 == v2
+	case GreaterThan:
+		return v1 > v2
+	case LessThan:
+		return v1 < v2
+	case LessThanInclusive:
+		return v1 <= v2
+	case GreaterThanInclusive:
+		return v1 >= v2
+	case NotEqual:
+		return v1 != v2
+	}
+	return v1 == v2
+}
+
+func matchFloat(c Operator, v1, v2 float64) bool {
+	switch c {
+	case Equal:
+		return v1 == v2
+	case GreaterThan:
+		return v1 > v2
+	case LessThan:
+		return v1 < v2
+	case LessThanInclusive:
+		return v1 <= v2
+	case GreaterThanInclusive:
+		return v1 >= v2
+	case NotEqual:
+		return v1 != v2
+	}
+	return v1 == v2
+}
+
+func matchString(c Operator, v1, v2 string) bool {
+	switch c {
+	case Contains:
+		return strings.Contains(v1, v2)
+	case NotContains:
+		return !strings.Contains(v1, v2)
+	case In:
+		return slices.Contains(strings.Split(v2, ","), v1)
```

Comment:
I don't see where in the code we account for the case when the context value is a list of values rather than an explodable string.

Related: https://github.com/Flagsmith/engine-test-data/issues/16

Reply:
It does now